flink/src/test/java/org/apache/iceberg/flink/FlinkTestBase.java

```diff
@@ -76,7 +99,7 @@ protected TableEnvironment getTableEnv() {
     TableResult tableResult = getTableEnv().executeSql(String.format(query, args));
     tableResult.getJobClient().ifPresent(c -> {
       try {
-        c.getJobExecutionResult(Thread.currentThread().getContextClassLoader()).get();
+        c.getJobExecutionResult().get();
```

Comment:
I read the javadoc about  [TableResult](https://github.com/apache/flink/blob/97bfd049951f8d52a2e0aed14265074c4255ead0/flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/TableResult.java#L186) again.  The correct way to execute sql and get the results is:

```java
TableResult result = tEnv.execute("select ...");
// using try-with-resources statement
try (CloseableIterator<Row> it = result.collect()) {
        it... // collect same data
}
```

Then I think we don't have to call `c.getJobExecutionResult().get()` here. How about removing  the line101 ~ line 106 and the following part use `try-with-resources` statement ?

Reply:
I was wondering the same question regarding `c.getJobExecutionResult().get()`. Updated the PR per your suggestion.

Reply:
BTW, the try-with-resource works well for Flink 1.12. It seems that it doesn't work with Flink 1.11 when I tried it in [the Flink test refactor PR](https://github.com/apache/iceberg/pull/2047)